aws/resources.go

```diff
@@ -427,6 +436,131 @@ func albs(ctx context.Context, a *aws, resourceType string, tags []tag.Tag) ([]p
 	return resources, nil
 }
 
+func albListeners(ctx context.Context, a *aws, resourceType string, tags []tag.Tag) ([]provider.Resource, error) {
+
+	ALBArns, err := getLoadBalancersV2Arns(ctx, a, ALB.String(), tags)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, alb := range ALBArns {
+
+		input := &elbv2.DescribeListenersInput{
+			LoadBalancerArn: awsSDK.String(alb),
+		}
+
+		albListeners, err := a.awsr.GetLoadBalancersV2Listeners(ctx, input)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, i := range albListeners.Listeners {
+			r, err := initializeResource(a, *i.ListenerArn, resourceType)
+			if err != nil {
+				return nil, err
+			}
+			resources = append(resources, r)
+		}
+	}
+
+	return resources, nil
+}
+
+func albListenerRules(ctx context.Context, a *aws, resourceType string, tags []tag.Tag) ([]provider.Resource, error) {
+
+	ALBListeners, err := getLoadBalancersV2ListenersArns(ctx, a, ALBListener.String(), tags)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, l := range ALBListeners {
+
+		input := &elbv2.DescribeRulesInput{
+			ListenerArn: awsSDK.String(l),
+		}
+
+		albListenerRules, err := a.awsr.GetLoadBalancersV2Rules(ctx, input)
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, i := range albListenerRules.Rules {
+
+			r, err := initializeResource(a, *i.RuleArn, resourceType)
+
+			if err != nil {
+				return nil, err
+			}
+
+			resources = append(resources, r)
+		}
+	}
+
+	return resources, nil
+}
+
+func albListenerCertificates(ctx context.Context, a *aws, resourceType string, tags []tag.Tag) ([]provider.Resource, error) {
+
+	ALBListeners, err := getLoadBalancersV2ListenersArns(ctx, a, ALBListener.String(), tags)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, l := range ALBListeners {
+
+		input := &elbv2.DescribeListenerCertificatesInput{
+			ListenerArn: awsSDK.String(l),
+		}
+
+		albListenerCertificates, err := a.awsr.GetListenerCertificates(ctx, input)
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, i := range albListenerCertificates.Certificates {
+
+			// TODO: if filter include aws_alb_listener, check if *i.IsDefault not append (since it is already written by aws_alb_listener)
+			r, err := initializeResource(a, *i.CertificateArn, resourceType)
+
+			if err != nil {
+				return nil, err
+			}
+
+			resources = append(resources, r)
+		}
+	}
+
+	//TODO: This resource it's not Importable yet (https://www.terraform.io/docs/providers/aws/r/lb_listener_certificate.html)
```

Comment:
I would add an issue and link it to this so we know/remember this if at somepoint it's importable.

If not, maybe we do not need this function as it has no use hehe.